lib/active_record/connection_adapters/cockroachdb/transaction_manager.rb

```diff
@@ -9,7 +9,7 @@ module TransactionManagerMonkeyPatch
         # transactions will be retried until they pass or the max retry limit is
         # exceeded.
         def within_new_transaction(isolation: nil, joinable: true, attempts: 0)
-          super
+          super(isolation: isolation, joinable: joinable)
```

Comment:
may be a noob question but why is attempts not passed through in this case

Reply:
Attempts is specific to cockroach. It used to be ok to pass it to `super` since the method signature was `within_new_transaction(options={})`, but now it's `within_new_transaction(isolation: nil, joinable: nil)` so passing `attempts` will raise an `ArgumentError`.

Reply:
ah, thanks!